[optimize](paimon): Avoid redundant hadoop conf setting in each split

suxiaogang223 opened the pull request:
### What problem does this PR solve?

Problem Summary:
Previously, PaimonScanNode.setPaimonParams() was called for each PaimonSplit and set hadoop conf (backendStorageProperties) in every split, which is redundant since these properties are the same at ScanNode level.

This commit:
1. Modifies PaimonJniReader to prefer hadoop conf from scan node level (range_params->properties) over split level, with fallback for backward compatibility
2. Removes redundant fileDesc.setHadoopConf() call in PaimonScanNode, since backendStorageProperties is already set at ScanNode level via params.properties in createScanRangeLocations()

This optimization:
- Reduces memory usage by avoiding duplicate configuration in each split
- Improves performance by eliminating redundant serialization
- Aligns with the implementation pattern used by Hudi and other data sources

### Release note

None

### Check List (For Author)

- Test <!-- At least one of them must be included. -->
    - [ ] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason <!-- Add your reason?  -->

- Behavior changed:
    - [ ] No.
    - [ ] Yes. <!-- Explain the behavior change -->

- Does this need documentation?
    - [ ] No.
    - [ ] Yes. <!-- Add document PR link here. eg: https://github.com/apache/doris-website/pull/1214 -->

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label <!-- Add branch pick label that this PR should merge into -->

hello-stephen commented:
Thank you for your contribution to Apache Doris.
Don't know what should be done next? See [How to process your PR](https://cwiki.apache.org/confluence/display/DORIS/How+to+process+your+PR).

Please clearly describe your PR:
1. What problem was fixed (it's best to include specific error reporting information). How it was fixed.
2. Which behaviors were modified. What was the previous behavior, what is it now, why was it modified, and what possible impacts might there be.
3. What features were added. Why was this function added?
4. Which code was refactored and why was this part of the code refactored?
5. Which functions were optimized and what is the difference before and after the optimization?

suxiaogang223 committed:
optimize(paimon): move hadoop conf and predicate to ScanNode level

Previously, both hadoop conf (backendStorageProperties) and paimon_predicate
were set in each PaimonSplit, causing redundant serialization when there are
many splits. This commit moves them to ScanNode level to avoid redundant
serialization.

Changes:
1. Added paimon_predicate field to TFileScanRangeParams in thrift
2. Modified PaimonJniReader to prefer hadoop conf and paimon_predicate from
   scan node level (range_params->properties and range_params->paimon_predicate)
   over split level, with fallback for backward compatibility
3. Set paimon_predicate at ScanNode level in PaimonScanNode.createScanRangeLocations()
4. Removed redundant fileDesc.setHadoopConf() and fileDesc.setPaimonPredicate()
   calls in setPaimonParams()

This optimization:
- Reduces memory usage by avoiding duplicate configuration and predicate
  serialization in each split
- Improves performance by eliminating redundant serialization
- Saves network bandwidth (e.g., 1KB predicate * 1000 splits = 1MB saved)
- Maintains backward compatibility with fallback to split level

suxiaogang223 commented:
run buildall

hello-stephen commented:
# Cloud UT Coverage Report
Increment line coverage ` ` :tada:

[Increment coverage report](http://coverage.selectdb-in.cc/coverage/c9e94c75f46034e8e89d904927c7f056cb1181c3_c9e94c75f46034e8e89d904927c7f056cb1181c3_cloud/increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/c9e94c75f46034e8e89d904927c7f056cb1181c3_c9e94c75f46034e8e89d904927c7f056cb1181c3_cloud/report/index.html)
| Category          | Coverage           |
|-------------------|--------------------|
| Function Coverage | 80.56% (1683/2089) |
| Line Coverage     | 66.73% (29695/44498) |
| Region Coverage   | 67.34% (14831/22024) |
| Branch Coverage   | 57.57% (7889/13704) |

hello-stephen commented:
# BE UT Coverage Report
Increment line coverage `0.00% (0/10)` :tada:

[Increment coverage report](http://coverage.selectdb-in.cc/coverage/c9e94c75f46034e8e89d904927c7f056cb1181c3_c9e94c75f46034e8e89d904927c7f056cb1181c3/increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/c9e94c75f46034e8e89d904927c7f056cb1181c3_c9e94c75f46034e8e89d904927c7f056cb1181c3/report/index.html)
| Category          | Coverage           |
|-------------------|--------------------|
| Function Coverage | 52.80% (18280/34618) |
| Line Coverage     | 38.17% (166194/435397) |
| Region Coverage   | 33.16% (129185/389570) |
| Branch Coverage   | 33.89% (55440/163568) |

hello-stephen commented:
# BE Regression && UT Coverage Report
Increment line coverage `0.00% (0/10)` :tada:

[Increment coverage report](http://coverage.selectdb-in.cc/coverage/57950_c9e94c75f46034e8e89d904927c7f056cb1181c3_merge/increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/57950_c9e94c75f46034e8e89d904927c7f056cb1181c3_merge/report/index.html)
| Category          | Coverage           |
|-------------------|--------------------|
| Function Coverage | 71.53% (24333/34018) |
| Line Coverage     | 58.01% (252978/436073) |
| Region Coverage   | 53.37% (210850/395081) |
| Branch Coverage   | 54.65% (89976/164643) |
